server/src/main/java/org/elasticsearch/transport/InboundHandler.java

```diff
@@ -260,6 +281,7 @@ private <T extends TransportRequest> void handleRequest(TcpChannel channel, Inbo
             final T request;
             try {
                 request = reg.newRequest(stream);
+                message.close(); // eager release message to save heap
```

Comment:
Could we drop this one line from this PR and do it in an immediate follow-up? That way we have one commit that replaces refcounting with one-shot closing without changing semantics and then another that reduces the lifecycle.

Reply:
++ makes sense dropped it :0